common/interfaces/card_interface/nfc_events.c

```diff
@@ -0,0 +1,154 @@
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "nfc_events.h"
+
+#include "memzero.h"
+#include "string.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define DEFAULT_NFC_TIMEOUT 100
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static nfc_ctx_t nfc_ctx;
+static nfc_event_t nfc_event;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+bool nfc_get_event(nfc_event_t *nfc_event_os_obj) {
+  ASSERT(nfc_event_os_obj != NULL);
+
+  if (nfc_event.event_occured == true) {
+    memcpy(nfc_event_os_obj, &nfc_event, sizeof(nfc_event_t));
+    nfc_reset_event();
+    return true;
+  }
+  return false;
+}
+
+void nfc_reset_event() {
+  memzero(&nfc_event, sizeof(nfc_event_t));
+}
+
+void nfc_enable_card_detect_event() {
+  nfc_ctx.card_detect_enabled = true;
+}
+
+void nfc_disable_card_detect_event() {
+  nfc_ctx.card_detect_enabled = false;
+}
+
+void nfc_set_card_detect_event() {
+  if (nfc_ctx.card_detect_enabled) {
+    nfc_event.event_occured = true;
+    nfc_event.event_type = NFC_EVENT_CARD_DETECT;
+  }
+}
+
+void nfc_tick_inc(uint16_t tick_inc) {
+  if (nfc_ctx.nfc_time < DEFAULT_NFC_TIMEOUT)
+    nfc_ctx.nfc_time += tick_inc;
+}
+
+void nfc_ctx_init() {
+  if (nfc_ctx.card_detect_enabled) {
+    nfc_ctx.nfc_time = 0;
+    nfc_ctx.nfc_field_off = false;
+  }
```

Comment:
this could be very confusing. this means the `nfc_enable_card_detect_event` needs to be called before `nfc_ctx_init` which is not very intuitive unless this function returns some response.

Reply:
Agreed, decided to remove enable disable requirements.